### PR TITLE
[`feat`] Integration of dataset benchmarks

### PR DESCRIPTION
**Pull request by ir2718**
Hi,

as mentioned in issue https://github.com/Topdu/OpenOCR/issues/24, this is a PR for introducing benchmark datasets for the STR task. Each of the datasets has a .yaml config defined which is used for the scripts. For example:
```
python tools/download/download_dataset.py --config configs/dataset/rec/iiit.yaml
python tools/create_lmdb_dataset.py --config configs/dataset/rec/iiit.yaml 
```
The create_lmdb_dataset.py script is slightly changed to accomodate for dynamically creating a lmdb dataset. The current PR does not support different splits (uses a concatenation of all examples). 

For now, I'll set this as a draft PR as I'm open to any suggestions or changes.

@Topdu

### CODE REVIEW

**Comment by Topdu** [comment]
Thank you for your efforts, it's a very exciting one to work on. For some of these datasets, they have been changed compared to the original versions, especially for commonly used datasets, e.g., PARSeq has made changes to some of the mislabeling in IIIT5k, SVT, SVTP, IC13, IC15, CUTE80. And many of the work follows the datasets provided by PARSeq, I suggest you to download [the datasets provided by PARSeq](https://github.com/baudm/parseq/tree/main?tab=readme-ov-file#datasets) directly, that many have been uploaded to Google Drive. For the training and test sets of Union14M-L we will provide a Google Drive version. We will let you know specifically at that time.

**Comment by Topdu** [comment]
### Downloading Datasets

All data can be downloaded from [Google Drive](https://drive.google.com/drive/folders/17mK_PLiBU0-a8qhUd5g4H6V03yUM2Gyp?usp=sharing).

#### The structure of Datasets and OpenOCR code will be organized as follows:
  <details close>
  <summary><strong>Structure of Datasets and OpenOCR code</strong></summary>

    ```text
    benchmark_bctr # Chinese text datasets, optional
    ├── benchmark_bctr_test
    │   ├── document_test
    │   ├── handwriting_test
    │   ├── scene_test
    │   └── web_test
    └── benchmark_bctr_train
        ├── document_train
        ├── handwriting_train
        ├── scene_train
        └── web_train
    evaluation
    ├── CUTE80
    ├── IC13_857
    ├── IC15_1811
    ├── IIIT5k
    ├── SVT
    └── SVTP
    iiit5k_test_images # for Latency Measurement, optional
    ltb # Long Text Benchmark
    OpenOCR
    OST
    synth # optional
    ├── MJ
    │   ├── test
    │   ├── train
    │   └── val
    └── ST
    test # Common Benchmarks from PARSeq
    ├── ArT
    ├── COCOv1.4
    ├── CUTE80
    ├── IC13_1015
    ├── IC13_1095  
    ├── IC13_857
    ├── IC15_1811
    ├── IC15_2077
    ├── IIIT5k
    ├── SVT
    ├── SVTP
    └── Uber
    u14m # lmdb format Union14M-Benchmark
    ├── artistic
    ├── contextless
    ├── curve
    ├── general
    ├── multi_oriented
    ├── multi_words
    └── salient
    Union14M-L-LMDB-Filtered # lmdb format Union14M-L-Filtered
    ├── train_challenging
    ├── train_easy
    ├── train_hard
    ├── train_medium
    └── train_normal
    ```
  </details>


#### **Datasets used during Training**

| Datsets | Google Drive | Baidu Yun|
| --- | --- | --- |
| Union14M-L-Filter | [LMDB archives](https://drive.google.com/drive/folders/1OlDWJZgvd6s4S09S3IGeAI90jI0i7AB_?usp=sharing) | |
| Evaluation | [LMDB archives](https://drive.google.com/drive/folders/1EW0_YvmRSdpVOkR2guTQFrGz7KNqNc66?usp=drive_link)| |

If you have downloaded Union14M-L, you can use [the filtered list of images](https://drive.google.com/drive/folders/1x1LC8C_W-Frl3sGV9i9_i_OD-bqNdodJ?usp=drive_link) to create an LMDB of the training set Union14M-L-Filter.

#### **Test Set**
| Datsets | Google Drive | Baidu Yun|
| --- | --- | --- |
| Union14M-L-Benchmark | [LMDB archives](https://drive.google.com/drive/folders/182RBQqjMVRZAXe0kpUJbMiDo7RWHwcSK?usp=drive_link) | |
| Common-Benchmarks | [LMDB archives](https://drive.google.com/drive/folders/103DA5Wu9YofDbE-JIUm5StJxTO2GixgS?usp=drive_link) | |
| Long Text Benchmark (LTB) | [LMDB archives](https://drive.google.com/drive/folders/1NChdlw7ustbXtlFBmh_0xnHvRkffb9Ge?usp=sharing) | |
| Occluded Scene Text (OST) | [LMDB archives](https://drive.google.com/drive/folders/1c0NVNF4yRACiP2IfeO_GnrHj0u_FhnpN?usp=drive_link) | |

**Note**: Both **Union14M-L-Filter** and **Union14M-L-Benchmark** are based on [Union14M-L](https://github.com/Mountchicken/Union14M?tab=readme-ov-file#3-union14m-dataset) and therefore comply with its copyright. Common Benchmarks and OST are derived from [PARSeq](https://github.com/baudm/parseq/tree/main?tab=readme-ov-file#datasets) and [VisionLAN](https://github.com/wangyuxin87/VisionLAN/blob/main/README.md#results-on-ost-datasets), respectively.

**Comment by ir2718** [comment]
Hi,

I've finally found time start wrapping up this PR. The uploaded datasets helped out immensely. If you've got time, @Topdu please have a look and tell me if you would like me to change something. 

Before converting this into a non-draft PR, I'd like to ask 2 questions:
1. Aside from the aforementioned datasets, I also added OpenVINO and TextOCR from parseq. However, I noticed parseq also uses LSVT, MLT19, RCTW17, ReCTS, and COCO-Text v2.0. Would you like me to add those as well?
2. I'm having trouble finding a link for the `benchmark_bctr` datasets as I'm not well acquainted with chinese text recognition. Can you provide me with a link for downloading those datasets?

**Comment by ir2718** [comment]
Are there any updates on the datasets? @Topdu

**Comment by Topdu** [comment]
Very sorry for our delayed response.

> 1. Aside from the aforementioned datasets, I also added OpenVINO and TextOCR from parseq. However, I noticed parseq also uses LSVT, MLT19, RCTW17, ReCTS, and COCO-Text v2.0. Would you like me to add those as well?

They are already included in Union14M, and you can leave them out for now.

> 2. I'm having trouble finding a link for the `benchmark_bctr` datasets as I'm not well acquainted with chinese text recognition. Can you provide me with a link for downloading those datasets?

Since benchmark_bctr is copyrighted (especially regarding the Chinese handwriting dataset), it is recommended to request permission to use it from the dataset's preliminary creator. After getting permission from the creator, we can send it to the applicant confidentially.


> Are there any updates on the datasets? @Topdu

I think this draft version can be merged as a ready PR.

Thank you for your outstanding contribution.

**Comment by ir2718** [comment]
You're welcome. In the future I hope I'll find time to work on more PRs.

File: docs/openocr.md [hunk 1]
```diff
@@ -80,6 +80,45 @@ tar xf OCR_e2e_img.tar
 python demo_gradio.py
 ```
 
+## Downloading Preprocessed Datasets
```
Comment (Topdu) [inline docs/openocr.md:83]: These datasets were used to train the research version of SVTRv2, there are other data in the OpenOCR dataset, you can remove this data description for now, and we will subsequently add instructions for using the configuration file to download the dataset in SVTRv2.

File: tools/create_lmdb_dataset.py [hunk 2]
```diff
@@ -5,6 +5,17 @@
 import numpy as np
 import io
 from PIL import Image
+import sys
```
Comment (Topdu) [inline tools/create_lmdb_dataset.py:8]: All datasets are in LMDB format and can be created without modifying create_lmdb_dataset.

Comment (ir2718) [reply tools/create_lmdb_dataset.py:8]: I used this for trying out some stuff locally, so I forgot to remove it. Thanks.

**Comment by Topdu** [comment]
It is now ready to be merged in. Thanks to your efforts!!

**Review by Topdu — APPROVED** [review]
Done
